### ISSUE TITLE
Fix doc tests that use the target triple other than the host

### ISSUE THREAD

**Issue by volks73** [volks73/cargo-rustc-cfg#6]
Three of the doc tests are currently ignored because I cannot get them to work/pass. I get an error that the "+nightly" subcommand is not found, but when I use the `dbg!` macro on the `cmd` just before execution, the command looks perfect. The command works when I type it manually into the terminal without any problems. Originally, I thought it was related to the `rustc_target` and `rustc_targets` methods, but when I commented this functionality out, it still occurred. I think it might be something with the doc test implementation.

**Comment by volks73** [comment]
Okay, it appears to be an issue with doc tests and "cross-compiling" not being allowed. If I switch the default toolchain from stable to nightly using rustup and remove the cfg attributes to allow the tests to run on any platform, the tests run, but fail for other reasons. The same error also happens when using the `host` function, too. Removing the "+nightly" from the command will remove the error, but I need to completely rethink the tests. The cfg attributes do not appear to be working to only compile the tests if the host matches the configuration attributes.

Basically, the doc tests and testing in general for the crate need to be refactored.

**Comment by volks73** [comment]
It appears that rustdoc ignores conditional compilation flags and honoring conditional compilation is an unresolved "thing". The test are being refactor to use platform-agnostic assertions when documenting host-specific functionality. The tests can be run with `cargo +nightly test`. For now, comments about requiring the nightly toolchain have been added.

**Comment by volks73** [comment]
This is fixed as of f297fbb26768d41894008d5a25706b2c81acc486. Different kinds of assertions are used for host-specific doc tests.
